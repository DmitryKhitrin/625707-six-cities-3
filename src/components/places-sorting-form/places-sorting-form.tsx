import React, {FC, memo} from "react";

type Props = {
  sortType: string;
  onSetSortType: (T: React.MouseEvent<HTMLLIElement, MouseEvent>) => void;
  onToggleSortMenu: () => void;
  isMenuOpen: boolean;
}

const PlacesSortingForm: FC<Props> = ({
  sortType = ``,
  onSetSortType = () => {},
  onToggleSortMenu = () => {},
  isMenuOpen = true,
}) => {
  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span
        className="places__sorting-type"
        tabIndex={0}
        onClick={onToggleSortMenu}
      >
        {sortType}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select" />
        </svg>
      </span>
      <ul
        className={`places__options places__options--custom ${
          isMenuOpen ? `places__options--opened` : ``
        }`}
      >
        <li
          className={`places__option ${
            sortType === `Popular` ? `places__option--active` : ``
          }`}
          tabIndex={0}
          onClick={onSetSortType}
        >
          Popular
        </li>
        <li
          className={`places__option ${
            sortType === `Price: low to high` ? `places__option--active` : ``
          }`}
          tabIndex={0}
          onClick={onSetSortType}
        >
          Price: low to high
        </li>
        <li
          className={`places__option ${
            sortType === `Price: high to low` ? `places__option--active` : ``
          }`}
          tabIndex={0}
          onClick={onSetSortType}
        >
          Price: high to low
        </li>
        <li
          className={`places__option ${
            sortType === `Top rated first` ? `places__option--active` : ``
          }`}
          tabIndex={0}
          onClick={onSetSortType}
        >
          Top rated first
        </li>
      </ul>
    </form>
  );
};

const MemoizedPlacesSortingForm = memo(PlacesSortingForm);
export {MemoizedPlacesSortingForm as PlacesSortingForm};
