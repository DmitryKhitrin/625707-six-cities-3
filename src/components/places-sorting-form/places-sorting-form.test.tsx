import React from "react";
import renderer from "react-test-renderer";
import {PlacesSortingForm} from "./places-sorting-form";

describe(`<PlacesSortingForm /> should render PlacesSortingForm.`, () => {
  it(`<PlacesSortingForm /> should render PlacesSortingForm with opened menu.`, () => {
    const tree = renderer
      .create(
          <PlacesSortingForm
            setSortType={jest.fn()}
            sortType={``}
            isMenuOpen={true}
            toggleSortMenu={jest.fn()}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`<PlacesSortingForm /> should render PlacesSortingForm with closed menu.`, () => {
    const tree = renderer
      .create(
          <PlacesSortingForm
            setSortType={jest.fn()}
            sortType={``}
            isMenuOpen={false}
            toggleSortMenu={jest.fn()}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
