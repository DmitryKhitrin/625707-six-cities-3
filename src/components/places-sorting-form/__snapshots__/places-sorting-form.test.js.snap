// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlacesSortingForm /> should render PlacesSortingForm. <PlacesSortingForm /> should render PlacesSortingForm with closed menu. 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex="0"
  >
    
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
  >
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`<PlacesSortingForm /> should render PlacesSortingForm. <PlacesSortingForm /> should render PlacesSortingForm with opened menu. 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex="0"
  >
    
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
  >
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      onClick={[MockFunction]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
